.github/workflows/build-test-publish.yml

```diff
@@ -0,0 +1,161 @@
+name: Build, Test, Publish
+
+on:
+  pull_request:
+  push:
+    branches: ["main"]
+  schedule:
+    - cron: '0 0 * * *'
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+
+  create:
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        builder: ["buildpacks-18", "buildpacks-20", "builder-classic-22", "builder-22"]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: buildpacks/github-actions/setup-pack@v4.8.1
+      - name: Setup builder cache
```

Comment:
I presume the use of the cache rather than the artifacts feature is due to the slowdowns mentioned on Slack? Would it be worth adding a comment explaining why we're not using what GitHub suggests for inter job persistence?

Also, one concern I had was about the potential for cache poisoning (eg from a PR), however it seems the jobs from other branches cannot write to the `main` branch, if I'm understanding this correctly (so we should be fine):
https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache

Reply:
Yes, this turned out to be 4x faster. Cache persistence and restoration was around 30s, where the upload-artifact was over 2 minutes. Also artifacts are available after the job completes, which is not something we care about here. 

My impression was that artifacts were better suited for build reports or production binaries, and caches are better suited for intra-workflow persistence.

I'll add a comment.